Ghpr.Core/Reporter.cs

```diff
@@ -1,10 +1,10 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Threading;
 using Ghpr.Core.Common;
 using Ghpr.Core.EmbeddedResources;
 using Ghpr.Core.Extensions;
 using Ghpr.Core.Interfaces;
-using Ghpr.Core.Utils;
 
 namespace Ghpr.Core
```

Comment:
All the dependencies should be injected via constructor and made private readonly. TestRunStrated can be set up in constructor as well. Also think of adding IOC Container such as Autofac or Unity